autoscaler/autoscaler.tf

```
@@ -46,6 +46,15 @@ resource "aws_lambda_function" "autoscaler" {
   architectures = [var.autoscaling_configuration.architecture == "amd64" ? "x86_64" : coalesce(var.autoscaling_configuration.architecture, "x86_64")]
   timeout       = var.autoscaling_configuration.timeout != null ? var.autoscaling_configuration.timeout : 30
 
+  dynamic "vpc_config" {
+    for_each = var.subnet_ids != null && var.security_group_ids != null ? [true] : []
```

The purpose of this construct is not obvious, at least to me. For clarity and consistency, we could use [the same pattern](https://github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2/blob/master/iam.tf#L121) as in the `iam.tf` file.
```suggestion
    for_each = var.subnet_ids != null && var.security_group_ids != null ? ["USE_VPC_CONFIG"] : []
```